src/policies/exponential_backoff.rs

```diff
@@ -244,8 +244,74 @@ impl ExponentialBackoffBuilder {
             },
         }
     }
-}
 
+    /// Builds an [`ExponentialBackoff`] with the given maximum total duration and calculates max
+    /// retries that should happen applying a 1.0 jitter factor.
+    /// We will enforce whatever comes first, max retries or total duration.
+    ///
+    /// Requires the use of [`ExponentialBackoffTimed::for_task_started_at()`].
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use chrono::{DateTime, Utc};
+    /// use retry_policies::{RetryDecision, RetryPolicy};
+    /// use retry_policies::policies::ExponentialBackoff;
+    /// use std::time::Duration;
+    ///
+    /// let exponential_backoff_timed = ExponentialBackoff::builder()
+    ///     .retry_bounds(Duration::from_secs(1), Duration::from_secs(6 * 60 * 60))
+    ///     .build_with_total_retry_duration_and_max_retries(Duration::from_secs(24 * 60 * 60));
+    ///
+    /// assert_eq!(exponential_backoff_timed.backoff.max_n_retries, Some(17));
+    ///
+    /// let started_at = Utc::now()
+    ///     .checked_sub_signed(chrono::Duration::seconds(25 * 60 * 60))
+    ///     .unwrap();
+    ///
+    /// exponential_backoff_timed.for_task_started_at(started_at)
+    ///     .should_retry(0); // RetryDecision::DoNotRetry
+    ///
+    /// let started_at = Utc::now()
+    ///     .checked_sub_signed(chrono::Duration::seconds(1 * 60 * 60))
+    ///     .unwrap();
+    ///
+    /// exponential_backoff_timed.for_task_started_at(started_at)
+    ///     .should_retry(18); // RetryDecision::DoNotRetry
+    /// ```
+    pub fn build_with_total_retry_duration_and_max_retries(
```

Comment:
I'm assuming this implements this:

![image](https://github.com/TrueLayer/retry-policies/assets/1884486/29c11b03-7b06-44b1-b243-314f3704c61f)

Rather than this:

![image](https://github.com/TrueLayer/retry-policies/assets/1884486/ad31c245-a145-44da-8a5d-ae8d9f4c83de)

If so, could we make this clear?

Reply:
Also, what are your thoughts on the effect of how long tasks take?

![image](https://github.com/TrueLayer/retry-policies/assets/1884486/9410cb4a-5722-4ee4-82e1-722d104ec9e7)

Reply:
> I'm assuming this implements this:

Exactly, I will make it clear.

Reply:
Cool, yeah that makes sense. I wasn't really sure what I was asking either! I figured it would work fine.

I was just aware that my mental model was originally assuming that task duration was negligible compared with retry intervals, which is often not true.